WordPress/gutenberg
# Allow blocks to use the role attribute

## What?

This PR:

- Adds a new `role` block support that allows developers to add a `role` attribute in the markup without causing block validation errors.
- Adds the role block support to the group block.

Closes https://github.com/WordPress/gutenberg/issues/46002

## Why?

Allowing the role attribute increases the flexibility for theme and plugin authors. It compliments the HTML element selection option that is in the Advanced panel for supported blocks.

## Testing Instructions

In your WordPress install, locate wp-includes/class-wp-block-supports.php.
On line 184, inside function `get_block_wrapper_attributes,`
add `role` to the array of attributes:
`$attributes_to_merge = array( 'style', 'class', 'id', 'role' );`

Next, login to your WordPress install and create a new post.
Add a group block with a list of links inside.

Open the code editor mode.
Add "role":"navigation" in the block comment.
Add `role="navigation"` to the div:
```

<div class="wp-block-group" role="navigation">
<ul>
<li><a href="#">one</a></li>



<li><a href="#">two</a></li>



<li><a href="#">three</a></li>
</ul>
</div>


```

Save and exit code editor mode.
Confirm that there are no block validation errors.
View source;  confirm that the role on the group block is still `role="document"`

View the post on the front, confirm that the role attribute is present.
This should be tested with and work with themes with and without theme.json.

To test with a dynamic block:
Locate and open the block.json file for a dynamic block, for example the avatar, latest posts or post featured image.
Add `"role":true` to the list of **block supports.**
Build the Gutenberg plugin to apply the change.

Add the block in the block editor.
Open the code editor mode and add the role to the block comment (see example above).
Save and confirm that there are no block validation errors.
View the block on the front and confirm that the role attribute is added correctly.

(Don't forget to revert the change you made to the WordPress file when you are done testing :) )
